lib/validator.js

```diff
@@ -17,37 +23,65 @@ var inMapping = {
  * @param parameters list of params
  * @returns JSON schema
  */
-function convertToJsonSchema(parameters) {
+Validator.prototype._convertToJsonSchema = function(parameters) {
     var schema = {
         type: 'object',
         properties: {}
     };
 
     parameters.forEach(function(param) {
-        if (!schema.properties[inMapping[param.in]]) {
-            schema.properties[inMapping[param.in]] = {
-                type: 'object',
-                properties: {}
-            };
-            // 'required' array must have at least one element according to json-schema spec,
-            // se we can't preinitialize it.
-            schema.required = schema.required || [];
-            schema.required.push(inMapping[param.in]);
-        }
-        var reqPartSchema = schema.properties[inMapping[param.in]];
-        reqPartSchema.properties[param.name] = {};
-        if (param.required) {
-            reqPartSchema.required = reqPartSchema.required || [];
-            reqPartSchema.required.push(param.name);
+        if (param.in !== 'body') {
+            if (!schema.properties[inMapping[param.in]]) {
+                schema.properties[inMapping[param.in]] = {
+                    type: 'object',
+                    properties: {}
+                };
+                // 'required' array must have at least one element according to json-schema spec,
+                // se we can't preinitialize it.
+                schema.required = schema.required || [];
+                schema.required.push(inMapping[param.in]);
+            }
+
+            var reqPartSchema = schema.properties[inMapping[param.in]];
+            var paramSchema = {};
+            // We can't type-check directly, because everything come in as a string
```

Comment:
An alternative to these string-based tests might be to attempt a coercion right away (using the rules set out below), followed by a validation against the actual schema types. The coercion could return errors of its own, so that we get the helpful error messages describing the correct syntax for string parameters.

Main advantage of this approach is a consistent coercion pass, which can further simplify the code consuming the post data. It would also systematically eliminate differences between multipart form submissions and JSON requests.